### ISSUE TITLE
Weak pointer functions do not check their arguments, leads to crash

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap#1001]
The following examples crash with a segmentation fault:
```
gap> ElmWPObj(1, 1);
gap> r := rec( x := 1); SetElmWPObj(r, 1, 1); r;
```
This was originally reported by @rbehrends at http://tracker.gap-system.org/issues/371

**Comment by ChrisJefferson** [comment]
Will have a look at this

**Comment by ChrisJefferson** [comment]
Fixed in #1009
